src/components/explore/PageControls.tsx

```diff
@@ -0,0 +1,44 @@
+import Link from "next/link";
+import type { FC } from "react";
+
+import { exploreQuery, newURL } from "~/utils/ExploreUrlMaker";
+
+interface props {
+  num_pages: number;
+  query: exploreQuery;
+}
+
+let PageControls: FC<props> = ({ num_pages: max, query }) => {
+  let page = query.page || 1;
+  let page_numbers = [page - 2, page - 1, page, page + 1, page + 2];
+
+  page_numbers = page_numbers.filter((e) => e <= max && e > 0);
+
+  let redirectTo = newURL(query);
```

Comment:
this is a function - but that isnt clear. The curried function thing is definitely questionable.